sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -188,18 +183,10 @@ case class DataSourceResolution(
       }
 
     case ShowTablesStatement(None, pattern) =>
-      defaultCatalog match {
-        case Some(catalog) =>
-          ShowTables(
-            catalog.asTableCatalog,
-            catalogManager.currentNamespace,
-            pattern)
-        case None =>
-          ShowTablesCommand(None, pattern)
-      }
+      ShowTables(currentCatalog.asTableCatalog, catalogManager.currentNamespace, pattern)
 
     case ShowTablesStatement(Some(namespace), pattern) =>
-      val CatalogNamespace(maybeCatalog, ns) = namespace
+      val DefaultCatalogAndNamespace(maybeCatalog, ns) = namespace
```

Comment:
ditto

Reply:
OK. I updated to use `CurrentCatalogAndNamespace`. This will not fall back to `ShowTablesCommand` and there will be no `isTemporary` column any longer. This should be OK, right?